sihl/src/sihl.mli

```diff
@@ -298,6 +298,90 @@ module Web : sig
     (** [find_exn request] returns the CSRF token of the current [request]. If
         the CSRF middleware is not installed, an exception is raised. *)
     val find_exn : Rock.Request.t -> string
+
+    module Crypto : sig
+      (** [token_length] is the amount of bytes used in the unencrypted CSRF
+          tokens. *)
+      val token_length : int
+
+      module Secret : sig
+        type t
+
+        (** [make raw] turns a [raw] string secret to a fixed length SHA256
+            digest. A fixed length secret is required for the hard-coded AES key
+            sizes to work. *)
+        val make : string -> t
+      end
+
+      (* TODO [aerben] how to not type this all out *)
```

Comment:
@joseferben Do you know of a way? The interface (except for the documentation) is pretty much duplicated

Reply:
Thx